compiler/src/dotty/tools/dotc/core/TypeErasure.scala

```diff
@@ -605,7 +605,9 @@ class TypeErasure(isJava: Boolean, semiEraseVCs: Boolean, isConstructor: Boolean
       case tp: TermRef =>
         sigName(tp.widen)
       case ExprType(rt) =>
-        sigName(defn.FunctionOf(Nil, rt))
+        // Before erasure the signature will be `=>X` and after erasure it will be `Function0`
```

Comment:
By definition, sigName tells you the signature of the type as it will be in bytecode, so it shouldn't depend on the phase. 
I think what you should do instead is to tweak the double-definition check logic to handle specially inline defs, probably around here: https://github.com/lampepfl/dotty/blob/bbc0df7d453f322b7477df6280dd76f216df26fa/compiler/src/dotty/tools/dotc/typer/Checking.scala#L868

Reply:
That does not work since you will still get MergeErrors.

Reply:
What does it even mean to merge denotations of inline methods ? We should only ever form MultiDenotation with them, which doesn't require merging.

Reply:
No, inline methods can override other methods, and can override each other. So we need a full merge.